swift-lang/src/main/java/fr/insideapp/sonarqube/swift/lang/antlr/HighlighterVisitor.java

```diff
@@ -0,0 +1,210 @@
+            Swift5Parser.HASH_FUNCTION, Swift5Parser.HASH_DSO_HANDLE, Swift5Parser.HASH_SELECTOR,
+            Swift5Parser.HASH_KEYPATH, Swift5Parser.HASH_COLOR_LITERAL, Swift5Parser.HASH_FILE_LITERAL,
+            Swift5Parser.HASH_IMAGE_LITERAL, Swift5Parser.GETTER, Swift5Parser.SETTER, Swift5Parser.Identifier,
+            Swift5Parser.DOT, Swift5Parser.LCURLY, Swift5Parser.LPAREN, Swift5Parser.LBRACK, Swift5Parser.RCURLY,
+            Swift5Parser.RPAREN, Swift5Parser.RBRACK, Swift5Parser.COMMA, Swift5Parser.COLON, Swift5Parser.SEMI,
+            Swift5Parser.LT, Swift5Parser.GT, Swift5Parser.UNDERSCORE, Swift5Parser.BANG, Swift5Parser.QUESTION,
+            Swift5Parser.AT, Swift5Parser.AND, Swift5Parser.SUB, Swift5Parser.EQUAL, Swift5Parser.OR, Swift5Parser.DIV,
+            Swift5Parser.ADD, Swift5Parser.MUL, Swift5Parser.MOD, Swift5Parser.CARET, Swift5Parser.TILDE,
+            Swift5Parser.HASH, Swift5Parser.BACKTICK, Swift5Parser.DOLLAR, Swift5Parser.BACKSLASH,
+            Swift5Parser.Operator_head_other, Swift5Parser.Operator_following_character, Swift5Parser.Binary_literal,
+            Swift5Parser.Octal_literal, Swift5Parser.Decimal_digits, Swift5Parser.Decimal_literal,
+            Swift5Parser.Hexadecimal_literal, Swift5Parser.Floating_point_literal, Swift5Parser.WS,
+            Swift5Parser.HASHBANG
+    );
+
+    @Override
+    public void apply(ParseTree tree) { // No default implementation
+    }
+
+    @Override
+    public void fillContext(SensorContext context, AntlrContext antlrContext) {
+        final InputFile file = antlrContext.getFile();
+        if (file == null) {
+            return;
+        }
+        final NewCpdTokens cpdTokens = context.newCpdTokens().onFile(file);
+        final NewHighlighting newHighlighting = context.newHighlighting().onFile(file);
+
+        for (final Token token : antlrContext.getTokens()) {
+            final int startLine = token.getLine();
+            final int startLineOffset = token.getCharPositionInLine();
+            final int[] endDetails = antlrContext.getLineAndColumn(token.getStopIndex());
+
+            if (endDetails == null
+                    || endDetails.length != 2
+                    || token.getType() == Recognizer.EOF
+                    || token.getType() == Swift5Parser.WS
+                    || token.getStartIndex() >= token.getStopIndex()) {
+                continue;
+            }
+
+            final int endLine = endDetails[0];
+            final int endLineOffset = endDetails[1] + (token.getText().contains("\n") ? 0 : 1);
+
+            try {
+                final TextRange range = file.newRange(startLine, startLineOffset, endLine, endLineOffset);
+                addHighlighting(newHighlighting, token, file, range, antlrContext.getVocabulary());
+                addCpdToken(cpdTokens, file, token, range);
+            } catch (final Exception e) {
+                if (LOGGER.isDebugEnabled()) {
```

Comment:
No need to test for debug.
You should also log the exception when using the scanner on production.